source/module_ri/Exx_LRI_interface.hpp

```diff
@@ -158,45 +161,50 @@ bool Exx_LRI_Interface<T, Tdata>::exx_after_converge(
             {
-                XC_Functional::set_xc_type(GlobalC::ucell.atoms[0].ncpp.xc_func);
+                restart_reset();
+                return true;
             }
+            else
+            {
+                this->etot_last_outer_loop = etot;
+                // update exx and redo scf
+                if (this->two_level_step == 0)
+                {
+                    XC_Functional::set_xc_type(GlobalC::ucell.atoms[0].ncpp.xc_func);
+                }
+
+                std::cout << " Updating EXX " << std::flush;
+                timeval t_start;       gettimeofday(&t_start, nullptr);
+
+                const bool flag_restart = (this->two_level_step == 0) ? true : false;
+                this->mix_DMk_2D.mix(dm.get_DMK_vector(), flag_restart);
+
+                // GlobalC::exx_lcao.cal_exx_elec(p_esolver->LOC, p_esolver->LOWF.wfc_k_grid);
+                const std::vector<std::map<int, std::map<std::pair<int, std::array<int, 3>>, RI::Tensor<Tdata>>>>
+                    Ds = GlobalV::GAMMA_ONLY_LOCAL
+                    ? RI_2D_Comm::split_m2D_ktoR<Tdata>(*this->exx_ptr->p_kv, this->mix_DMk_2D.get_DMk_gamma_out(), *dm.get_paraV_pointer(), GlobalV::NSPIN)
+                    : RI_2D_Comm::split_m2D_ktoR<Tdata>(*this->exx_ptr->p_kv, this->mix_DMk_2D.get_DMk_k_out(), *dm.get_paraV_pointer(), GlobalV::NSPIN);
+                this->exx_ptr->cal_exx_elec(Ds, *dm.get_paraV_pointer());
+                iter = 0;
+                this->two_level_step++;
 
-            std::cout << " Updating EXX " << std::flush;
-            timeval t_start;       gettimeofday(&t_start, nullptr);
-
-            const bool flag_restart = (this->two_level_step == 0) ? true : false;
-            this->mix_DMk_2D.mix(dm.get_DMK_vector(), flag_restart);
-
-            // GlobalC::exx_lcao.cal_exx_elec(p_esolver->LOC, p_esolver->LOWF.wfc_k_grid);
-			const std::vector<std::map<int,std::map<std::pair<int, std::array<int, 3>>,RI::Tensor<Tdata>>>>
-				Ds = GlobalV::GAMMA_ONLY_LOCAL
-                ? RI_2D_Comm::split_m2D_ktoR<Tdata>(*this->exx_ptr->p_kv, this->mix_DMk_2D.get_DMk_gamma_out(), *dm.get_paraV_pointer(), GlobalV::NSPIN)
-                : RI_2D_Comm::split_m2D_ktoR<Tdata>(*this->exx_ptr->p_kv, this->mix_DMk_2D.get_DMk_k_out(), *dm.get_paraV_pointer(), GlobalV::NSPIN);
-            this->exx_ptr->cal_exx_elec(Ds, *dm.get_paraV_pointer());
-            iter = 0;
-            this->two_level_step++;
-            
-            timeval t_end;       gettimeofday(&t_end, nullptr);
-            std::cout << "and rerun SCF\t"
-                << std::setprecision(3) << std::setiosflags(std::ios::scientific)
-                << (double)(t_end.tv_sec-t_start.tv_sec) + (double)(t_end.tv_usec-t_start.tv_usec)/1000000.0 
-                << std::defaultfloat << " (s)" << std::endl;
-            return false;
+                timeval t_end;       gettimeofday(&t_end, nullptr);
```

Comment:
initialize the variable t_end

Reply:
This is a standard struct in `sys/time.h` that cannot be directly assigned.